feat: implement jump between footnotes

paulovareiro29 opened the pull request:
<!-- ## Title: Please consider adding the [skip chromatic] flag to the PR title in case you don't need chromatic testing your changes. -->
## Description:
<!-- *PR notes: Please describe the changes in this PR.* -->
Closes #1772

## Definition of Reviewable:
- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

MartaPintoTeixeira reviewed (changes requested):
1 ) On the 1st sample make sure copy in storybook matches figma
2 ) On the 1st sample can me have the footnotes numbers in primary? or no because they are not links?
3 ) On inverted sample the footnotes number is white and copy primary-400 (please change it)
4 ) For the tmplt let @coraliefeil reply to my comment so we know if we keep tmplt as it is or add more copy so the anrchor (when click) is visible.

More feedback will be provided in chromatic

paulovareiro29 commented:
> 1 ) On the 1st sample make sure copy in storybook matches figma

Updated

> 2 ) On the 1st sample can me have the footnotes numbers in primary? or no because they are not links?

Only links are colored in primary, just as in the footnotes specification.

> 3 ) On inverted sample the footnotes number is white and copy primary-400 (please change it)

It is already like that when the footnotes number is a link. According to the figma specification, when there is no link (just number), then it should be the same color.

> 4 ) For the tmplt let @coraliefeil reply to my comment so we know if we keep tmplt as it is or add more copy so the anrchor (when click) is visible.

Alright

> More feedback will be provided in chromatic

FIxed

MartaPintoTeixeira reviewed (approved):
I am aproving this PR but want to say that once again in chromatic there was a "bug" with "52 Changes" to review and in this ticket it should only be "sd-footnotes" (1).

MartaPintoTeixeira commented:
Please change the healdline copy according to figma: "The Power of a Solid Design System"

MartaPintoTeixeira reviewed (dismissed):
After solving headline copy is changed in tmplt consider it approved from my side.
